src/connectors/components/OnboardingPagingDotsConnector.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+
+import { connect } from 'react-redux'
+
+// import * as Constants from '../../constants/indexConstants.js'
+import type { Dispatch, State } from '../../modules/ReduxTypes'
+import PagingDots from '../../modules/UI/components/PagingDots/PagingDots.js'
```

Comment:
Please change the `PagingDots.js` actual filename to `PagingDots.ui.js`

Reply:
Changed it to PagingDotsComponent Since that is the name inside the file. export default class PagingDotsComponent

Reply:
Allen, this is a classic example of a component that should use our discussed architecture of sending redux state via a selector function and not directly connecting the component to redux since it is a child component of a scene. This would prevent the need for a separate file altogether.

Reply:
Proposed style for a "semi-connected" child component

Use PureComponent instead of Component

Have parent pass in selector functions into the child component as props

In the child component `getDerivedStateFromProps()`, call the props functions and assign local component state the resulting values

ie.

```
static getDerivedStateFromProps(nextProps, prevState) {
  const newState = {
    totalItems: nextProps.getTotalItems()
    currentIndex: nextProps.getCurrencyIndex()
  }
}
```
As a PureComponent, you will not rerender if newState doesn't differ from previous state and React will shallow compare for you. No need to have child be connected to redux